data/block/block.go

```diff
@@ -581,3 +582,36 @@ func (h *Header) GetAdditionalData() headerVersionData.HeaderAdditionalData {
 	// no extra data for the initial version of shard block header
 	return nil
 }
+
+// CheckFieldsForNil checks a predefined set of fields for nil values
+func (h *Header) CheckFieldsForNil() error {
+	if h == nil {
+		return data.ErrNilPointerReceiver
+	}
+	if h.PrevHash == nil {
+		return fmt.Errorf("%w in Header.PrevHash", data.ErrNilValue)
+	}
+	if h.PrevRandSeed == nil {
+		return fmt.Errorf("%w in Header.PrevRandSeed", data.ErrNilValue)
+	}
+	if h.RandSeed == nil {
+		return fmt.Errorf("%w in Header.RandSeed", data.ErrNilValue)
+	}
+	if h.RootHash == nil {
+		return fmt.Errorf("%w in Header.RootHash", data.ErrNilValue)
+	}
+	if h.ChainID == nil {
+		return fmt.Errorf("%w in Header.ChainID", data.ErrNilValue)
+	}
+	if h.SoftwareVersion == nil {
```

Comment:
this one don't remember if it was populated from the start, (genesis?)

Reply:
I will check and come back with an answer to this. As far as I remember, we've populated this from the beginning.

Reply:
It is populated with an empty slice. As far as I've tested, the genesis + a few hundred blocks afterward, and everything went ok. Started the import-db process nevertheless.